### DISCUSSION TITLE
When using symlink edit source button leads to wrong source!

### DISCUSSION THREAD

**Discussion by ddranic** [squidfunk/mkdocs-material#7447] [Q&A]
Hello!
I have 2 sections:
- Public API
- API

PublicAPI has BaseMiddleware.md and API has the same one. So the decision was to use symlink. 
api/ folder is having all the objects in API so public API refers to some of them.

The symlink is:
public_api/ -> api/

The problem is that in Public API section all edit source buttons lead to public_api/BaseMiddleware.md which is considered wrong by github(no such file). Is it possible to make this buttons lead to api/BaseMiddleware.md instead?
![image](https://github.com/user-attachments/assets/1b9b8ced-b07b-45bf-a9b2-a86aaadd9080)

**Comment by alexvoss** [comment]
I may be wrong but for this to work, MkDocs would need to recognise a file as a symbolic link and give it special handling. I am not aware of such functionality. You could try to suggest it in the [MkDocs discussions](https://github.com/mkdocs/mkdocs/discussions).

A way to solve this yourself would be to create a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks) that re-writes the edit URL for those symlinks. I am not sure how many people have the same problem but if you come up with a generic solution, this might make a good contribution for [show & tell](https://github.com/squidfunk/mkdocs-material/discussions/categories/show-and-tell)?!
